### PR TITLE
Add webdispatcher module

### PR DESCRIPTION
**Pull request by rukawata**
## Problem:
No webdispatcher at the moment.

## Solution:
Add code to deploy webdispatcher in app tier.

## Test:
1. Deploy with below definition in input JSON and finished fine:
```
"application": {
    "sid": "HN1",
    "enable_deployment": true,
    "scs_instance_number": "01",
    "ers_instance_number": "02",
    "scs_high_availability": false,
    "application_server_count": 1,
    "webdispatcher_count": 2,
    "authentication": {
      "type": "key",
      "username": "azureadm"
    },
    "os": {
      "publisher": "suse",
      "offer": "sles-sap-12-sp5",
      "sku": "gen1"
    }
  },
```
1. Change the number of `webdispatcher_count`, see inventory ini/yml changed accordingly locally and remotely on RTI

### CODE REVIEW

File: deploy/terraform/modules/app_tier/nsg-webdisp.tf [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# NSG rule to deny internet access
+resource azurerm_network_security_rule webRule_internet {
+  count                       = local.enable_deployment ? 1 : 0
+  name                        = "Internert"
```
Comment (peterchoo) [inline deploy/terraform/modules/app_tier/nsg-webdisp.tf:4]:
```suggestion
  name                        = "Internet"
```

File: deploy/terraform/modules/app_tier/vm-webdisp.tf [hunk 2]
```diff
@@ -0,0 +1,112 @@
+# Create Web dispatcher NICs
+resource "azurerm_network_interface" "web" {
+  count                         = local.enable_deployment ? var.application.webdispatcher_count : 0
+  name                          = "${upper(var.application.sid)}_web${format("%02d", count.index)}-nic"
+  location                      = var.resource-group[0].location
+  resource_group_name           = var.resource-group[0].name
+  enable_accelerated_networking = local.web_nic_accelerated_networking
+
+  ip_configuration {
+    name                          = "ipconfig1"
```
Comment (YunziZhangAzure) [inline deploy/terraform/modules/app_tier/vm-webdisp.tf:10]:
```suggestion
    name                          = "IPConfig1"
```
Maybe make the name consistent with vm-app/vm-scs 's

https://github.com/Azure/sap-hana/blob/4f83a659355d5b7b7ed3aeb8b454202111f015c9/deploy/terraform/modules/app_tier/vm-app.tf#L10

File: deploy/terraform/modules/app_tier/vm-webdisp.tf [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  name                         = "${upper(var.application.sid)}_web-avset"
+  location                     = var.resource-group[0].location
+  resource_group_name          = var.resource-group[0].name
+  platform_update_domain_count = 20
+  platform_fault_domain_count  = 2
+  proximity_placement_group_id = lookup(var.infrastructure, "ppg", false) != false ? (var.ppg[0].id) : null
+  managed                      = true
+}
+
+# Create the Web dispatcher Load Balancer
```
Comment (YunziZhangAzure) [inline deploy/terraform/modules/app_tier/vm-webdisp.tf:29]: SCS's load balancer and availability set configuration all go into infrastructure.tf. Is it better to put webdispatcher's lb and avset into the infrastructure.tf as well to make them consistent?

File: deploy/terraform/modules/output_files/ansible_inventory.yml.tmpl [hunk 4]
```diff
@@ -75,6 +75,14 @@ all:
...
 
+    web:
+      hosts:
+%{~ for ip-web in ips-web }
+        ${ip-web}:
+          ansible_connection: "ssh"
+          ansible_user:       "${application.authentication.username}"
+%{~ endfor }
+
     # Groups below are collections of the above groups for easier refernce
```
Comment (YunziZhangAzure) [inline deploy/terraform/modules/output_files/ansible_inventory.yml.tmpl:86]:
```suggestion
    # Groups below are collections of the above groups for easier reference
```

File: deploy/terraform/modules/app_tier/vm-webdisp.tf (hunk 3 above)
Comment (rukawata) [reply deploy/terraform/modules/app_tier/vm-webdisp.tf]: Sure! Moved and add comments in infrastructure.tf.

**Pull request merged**